localstack/aws/protocol/service_router.py

```diff
@@ -275,28 +276,32 @@ def get_service_catalog() -> ServiceCatalog:
         return ServiceCatalog()
 
 
-def resolve_conflicts(candidates: Set[str], request: Request):
+def resolve_conflicts(candidates: Set[ServiceModel], request: Request):
     """
     Some service definitions are overlapping to a point where they are _not_ distinguishable at all
     (f.e. ``DescribeEndpints`` in timestream-query and timestream-write).
     These conflicts need to be resolved manually.
     """
-    if candidates == {"timestream-query", "timestream-write"}:
+    service_name_candidates = {service.service_name for service in candidates}
+    if service_name_candidates == {"timestream-query", "timestream-write"}:
         return "timestream-query"
-    if candidates == {"docdb", "neptune", "rds"}:
+    if service_name_candidates == {"docdb", "neptune", "rds"}:
         return "rds"
-    if candidates == {"sqs-query", "sqs"}:
+    if service_name_candidates == {"sqs"}:
         # SQS now have 2 different specs for `query` and `json` protocol. From our current implementation with the
         # parser and serializer, we need to have 2 different service names for them, but they share one provider
-        # implementation. `sqs-query` represents the legacy `query` protocol spec, and `sqs` the new `json` spec,
-        # default in botocore starting 1.31.81.
+        # implementation. `sqs-json` represents the `json` protocol spec, and `sqs` the `query` protocol
+        # (default again in botocore starting with 1.32.6).
         # The `application/x-amz-json-1.0` header is mandatory for requests targeting SQS with the `json` protocol. We
-        # can safely route them to the `sqs` JSON parser/serializer. If not present, route the request to `query`.
+        # can safely route them to the `sqs-json` JSON parser/serializer. If not present, route the request to the
+        # default sqs protocol (`query`).
         content_type = request.headers.get("Content-Type")
-        return "sqs" if content_type == "application/x-amz-json-1.0" else "sqs-query"
+        return "sqs-json" if content_type == "application/x-amz-json-1.0" else "sqs"
 
 
-def determine_aws_service_name(request: Request, services: ServiceCatalog = None) -> Optional[str]:
+def determine_aws_service_model(
+    request: Request, services: ServiceCatalog = None
+) -> Optional[ServiceModel]:
```

Comment:
I see your point. In this case the function is not used anywhere else in the whole organization, which is why I would like to avoid using a deprecation path.